chore(testproxy): Mutate rows bug fix

danieljbruce opened the pull request:
This PR makes a change to the test proxy that allows the test proxy to use the veneer layer instead to make its call to mutateRows. Using the veneer layer to make the call allows additional parsing to be done on the request and the results which ensures the right results are returned to the test runner enabling the test runner to pass its test.

The test in question is TestMutateRows_NoRetry_NonTransientErrors.

danieljbruce committed:
Change mutate to match veneer

Changed the test proxy functionality to match what is in the veneer layer and now the test case in question passes with a slight change in the grpc code returned.

danieljbruce committed:
Shorten the parsing of entries in the test proxy

This commit just permits a more concise expression for the entries that should be returned back to the test runner.

danieljbruce committed:
Merge branch 'main' of https://github.com/googleapis/nodejs-bigtable into mutate-rows-bug-fix

danieljbruce committed:
Merge branch 'main' into mutate-rows-bug-fix

ruyadorno commented on testproxy/services/bulk-mutate-rows.js:
```diff
@@ -14,38 +14,44 @@
+        entry: [],
+      };
+    } catch (error) {
+      if (error.name === 'PartialFailureError') {
+        return {
+          status: {code: grpc.status.OK, message: error.message},
+          entry: Array.from(error.errors.entries()).map(([index, entry]) => ({
+            index: index + 1,
+            status: entry,
+          })),
+        };
```
why is the proxy turning partial failures into an "OK" response here?

danieljbruce replied:
It meets the needs of the test case and is required for the test case to pass, but this is a good start to a discussion about what the test proxy *should* do.

danieljbruce replied:
Maybe https://github.com/googleapis/nodejs-bigtable/blob/000a353f05fa65bc0a7dc551c777ac55c6c08875/src/table.ts#L1540 should change, but then we have a breaking change?

liujiongxin replied:
We shouldn't change the node js client. 

You can return the status as is, just ensure the entries are filled correctly. I will update the test case as well as the guide of test proxy writeup.

danieljbruce committed:
Changes to test proxy proto

The test proxy uses a new test path now so these adjustments must be made to work with the test runner properly.

danieljbruce committed:
Revert "Changes to test proxy proto"

This reverts commit efbdadbfdcb96d4135d211617207ed2524648b04.

danieljbruce committed:
Merge branch 'mutate-rows-bug-fix' of https://github.com/danieljbruce/nodejs-bigtable into mutate-rows-bug-fix

danieljbruce committed:
Pass error along as is

As per the online discussion, we should simply pass along the error as is so that the corresponding error message and status can be reported.

danieljbruce committed:
Merge branch 'main' of https://github.com/googleapis/nodejs-bigtable into mutate-rows-bug-fix

ruyadorno reviewed (approved):
LGTM 👍
